### PR TITLE
Core/InstanceScript: Refactored door types to reflect its behavior

### PR DESCRIPTION
**Pull request by mdX7**
**Tests performed:**
tested ingame
(Does it build, tested in-game, etc.)

### CODE REVIEW

File: src/server/game/Instances/InstanceScript.cpp [hunk 1]
```diff
@@ -228,17 +228,20 @@ void InstanceScript::UpdateDoorState(GameObject* door)
...
                 break;
-            case DOOR_TYPE_PASSAGE:
+            case EncounterDoorBehavior::OpenWhenDone:
                 open = (info.bossInfo->state == DONE);
                 break;
-            case DOOR_TYPE_SPAWN_HOLE:
+            case EncounterDoorBehavior::OpenWhenInProgress:
                 open = (info.bossInfo->state == IN_PROGRESS);
                 break;
+            case EncounterDoorBehavior::OpenWhenNotDone:
```
Comment (Nyr97) [inline src/server/game/Instances/InstanceScript.cpp:242]: Out of curiosity, where did this type come from?

File: src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp [hunk 2]
```diff
@@ -346,7 +346,7 @@ class boss_halion : public CreatureScript
                             events.ScheduleEvent(EVENT_BREATH, randtime(Seconds(16), Seconds(25)));
                             break;
                         case EVENT_ACTIVATE_FIREWALL:
-                            // Flame ring is activated 5 seconds after starting encounter, DOOR_TYPE_ROOM is only instant.
+                            // Flame ring is activated 5 seconds after starting encounter, EncounterStateForOpenDoor::NotInProgress is only instant.
```
Comment (Nyr97) [inline src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:349]: In retail, normally doors that close in retail for a boss fight take 5s to close. Why is this any different?

File: src/server/game/Instances/InstanceScript.cpp (hunk 1 above)
Comment (mdX7) [reply src/server/game/Instances/InstanceScript.cpp:242]: for anduin bridge after encounter end. Bridges work opposite way, open = you cannot pass, closed = you can pass

File: src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp (hunk 2 above)
Comment (mdX7) [reply src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp:349]: out of scope

File: src/server/game/Instances/InstanceScript.cpp (hunk 1 above)
Comment (Nyr97) [reply src/server/game/Instances/InstanceScript.cpp:242]: They really like entangling stuff.

**Pull request merged**